torch/csrc/api/include/torch/nn/options/transformercoder.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <torch/arg.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/types.h>
+#include <torch/enum.h>
+
+#include <torch/nn/modules/transformerlayer.h>
+#include <torch/nn/modules/container/any.h>
+
+namespace torch {
+namespace nn {
+
+  /// Options for the `TransformerEncoder`
+  ///
+  /// Example:
+  /// ```
+  /// TransformerEncoderLayer encoderLayer(TransformerEncoderLayerOptions(512, 8).dropout(0.1));
+  /// auto options = TransformerEncoderOptions(encoderLayer, 6).norm(LayerNorm(LayerNormOptions({2})));
+  /// ```
+  struct TORCH_API TransformerEncoderOptions {
+    // This constructor will keep the a ref of passed in encoder_layer, so it keeps all the data in encoder_layer.
+    TransformerEncoderOptions(TransformerEncoderLayer encoder_layer, int64_t num_layers);
```

Comment:
nit: "will keep the a ref of passed in" -> "will keep a ref of passed in"

Actually, it's not really that it is keeping a reference, it's more that it takes ownership of the thing passed-by-value, right?

Reply:
I will make it more clear, the reason I say it is still a ref because they share the same storage

Reply:
Changed to `shallow copy`